### PR TITLE
Fs 2780 flag for multiple sections

### PR DESCRIPTION
**Pull request by RamuniN**
https://digital.dclg.gov.uk/jira/browse/FS-2780

### Description
Flag for multiple sections
* Implemented checkboxes for multiple sections to flag
* Updates flag banner to show multiple sections of flag
* Backward compatible with earlier flagged sections data.

Please also review associated PR https://github.com/communitiesuk/funding-service-design-assessment-store/pull/185

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### Screenshots of UI changes (if applicable)
image
image

### CODE REVIEW

File: app/assess/templates/macros/assessment_flag.html [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
-        <p class="govuk-body">{{ flag.section_to_flag }}</p>
+        <h2 class="govuk-heading-m">Section(s) flagged</h2>
+        {% for sub_section_id in flag.sections_to_flag %}
+            {% set sub_section_data = state.get_section_from_sub_criteria_id(sub_section_id) %}
+            {% if sub_section_data %}
+                <p class="govuk-body"><a class="govuk-link govuk-link--no-visited-state" href="{{ url_for('assess_bp.display_sub_criteria',application_id=application_id ,sub_criteria_id=sub_section_id) }}" target="_blank">{{ sub_section_data["sub_section_name"] }}</a> ({{ sub_section_data["parent_section_name"] }}) (Opens in new tab)</p>
+            {% else %}
+                <p class="govuk-body">{{ sub_section_id }}</p>
+            {% endif %}
+        {% endfor %}
```
Comment (tferns) [inline app/assess/templates/macros/assessment_flag.html:16]: Ah okay, nice work on this - looks like it handles the fallback well

**Review by tferns — DISMISSED** [review]
Just a minor question, but changes look good

**Review by tferns — APPROVED** [review]
Ah it was a typo! Thought It'd looked weird

File: app/assess/templates/macros/assessment_flag.html (hunk 1 above)
Comment (RamuniN) [reply app/assess/templates/macros/assessment_flag.html:16]: yeah. For the older flagged sections there is no href so we just display like before

File: app/assess/routes.py [hunk 2]
```diff
@@ -233,7 +233,18 @@ def score(
...
 def flag(application_id):
-    form = FlagApplicationForm()
+
+    # Get assessor tasks list
+    assessor_task_list_metadata = get_assessor_task_list_state(application_id)
+    fund = get_fund(assessor_task_list_metadata["fund_id"])
+    assessor_task_list_metadata["fund_name"] = fund.name
+    state = AssessorTaskList.from_json(assessor_task_list_metadata)
+    choices = [
+        (item["sub_section_id"], item["sub_section_id"])
```
[the remark this answers is not in the document]
Comment (RamuniN) [reply app/assess/routes.py]: I had this code implemented few weeks back(before teams are merged). I think when I used `(item["sub_section_id"], item["sub_section_name"])` there was a bug which I don't remember now. But surprisingly I don't see it anymore. So removing this from tuple. Good shout again.

**Pull request merged**